context/AuthContextProvider.tsx

```diff
@@ -0,0 +1,107 @@
+'use client';
+import { createContext, useContext, useEffect, useMemo, useState } from 'react';
+import { account } from '@/api/config';
+import { useRouter } from 'next/navigation';
+
+type UserCredentials = {
```

Comment:
Why `type` and not `interface`?

Reply:
I find it easier to extend types than interfaces especially with unions .... but if I need to use Interface I can :)

Reply:
There's not a big different in one or the other. Interfaces are used more to define objects. Types can define many things.